app/services/game-overlay/index.ts

```diff
@@ -0,0 +1,396 @@
+    //   'overlay',
+    // );
+  }
+
+  createPreviewWindows() {
+    this.previewWindows.recentEvents = this.windowsService.createOneOffWindowForOverlay({
+      ...this.commonWindowOptions,
+      width: 600,
+      transparent: true,
+      webPreferences: { offscreen: false },
+      isFullScreen: true,
+      componentName: 'OverlayPlaceholder',
+      title: $t('Recent Events'),
+    });
+
+    this.previewWindows.chat = this.windowsService.createOneOffWindowForOverlay({
+      ...this.commonWindowOptions,
+      height: 600,
+      transparent: true,
+      webPreferences: { offscreen: false },
+      isFullScreen: true,
+      componentName: 'OverlayPlaceholder',
+      title: $t('Chat'),
+    });
+  }
+
+  async configureWindows() {
+    Object.keys(this.windows).forEach((key: string) => {
+      const win = this.windows[key];
+      win.webContents.once('did-finish-load', () => this.onWindowsReady.next(win));
+
+      const position = this.determineStartPosition(key);
+      const size = key === 'chat' ? { width: 300, height: 600 } : { width: 600, height: 300 };
+      win.setBounds({ ...position, ...size });
+      this.previewWindows[key].setBounds({ ...position, ...size });
+    });
+
+    this.windows.recentEvents.loadURL(this.userService.recentEventsUrl());
+    this.windows.chat.loadURL(
+      await getPlatformService(this.userService.platform.type).getChatUrl(
+        this.customizationService.isDarkTheme ? 'night' : 'day',
+      ),
+    );
+
+    // this.windows.overlayControls.webContents.once('dom-ready', async () => {
+    //   this.onWindowsReady.next(this.windows.overlayControls);
+    // });
+  }
+
+  determineStartPosition(window: string) {
+    const pos = this.state.windowProperties[window].position;
+    if (pos) {
+      const display = electron.screen.getAllDisplays().find(display => {
+        const bounds = display.bounds;
+        const intBounds = pos.x > bounds.x && pos.y > bounds.y;
+        const extBounds = pos.x < bounds.x + bounds.width && pos.y < bounds.y + bounds.height;
+        return intBounds && extBounds;
+      });
+      if (display) return pos;
+    }
+    return this.defaultPosition(window);
+  }
+
+  resetPosition() {
+    Object.keys(this.windows).forEach((key: string) => {
+      const overlayId = this.state.windowProperties[key].id;
+      if (!overlayId) return;
+      const pos = this.defaultPosition(key);
+      const size = key === 'chat' ? { width: 300, height: 600 } : { width: 600, height: 300 };
+      this.windows[key].setBounds({ ...pos, ...size });
+      this.previewWindows[key].setBounds({ ...pos, ...size });
+      overlay.setPosition(overlayId, pos.x, pos.y, size.width, size.height);
+    });
+  }
+
+  private defaultPosition(key: string) {
+    this.SET_WINDOW_POSITION(key, null);
+    const [containerX, containerY] = this.getWindowContainerStartingPosition();
+    const x = key === 'recentEvents' ? containerX - 600 : containerX;
+    return { x, y: containerY };
+  }
```

Comment:
It feels like the above 3 functions: `determineStartPosition`, `resetPosition` and `defaultPosition` could be a little clearer if they used newlines to break things up a bit.